### DISCUSSION TITLE
Change umbracoKeyValue SQL type to nvarchar(max)

### DISCUSSION THREAD

**Discussion by Rockerby** [umbraco/Umbraco-CMS#14007] [Features and ideas]
Currently, the database table umbracoKeyValue has a value type of nvarchar(255) which limits the amount of data you can put in there. Could this be increased to nvarchar(max) to store more data?

Use cases for this could be for custom development where you want to store oAuth response objects, dynamic configuration (change during runtime) or for storing JSON data, to name a few. The current alternatives if you want to store more than 255 chars include creating your own custom data storage (like file-based or custom database tables).

This also causes potential issues if you develop using SQL Lite as that doesn't have the 255 constraint but will then break when migrating to a production SQL server.

**Comment by lassefredslund** [comment]
Hi @Rockerby.
I think it would be totally OK if this field was nvarchar(max).
I'll add it as "up for grabs".

**Comment by Rockerby** [comment]
Thanks @lassefredslund 

Just to keep this up to date, change made and [PR 14085 created](https://github.com/umbraco/Umbraco-CMS/pull/14085)
